ddtrace/appsec/_asm_request_context.py

```diff
@@ -261,7 +261,7 @@ def get_headers_case_sensitive() -> bool:
     return get_value(_WAF_ADDRESSES, SPAN_DATA_NAMES.REQUEST_HEADERS_NO_COOKIES_CASE, False)  # type : ignore
 
 
-def set_block_request_callable(_callable: Optional[Callable]) -> None:
+def set_block_request_callable(_callable: Optional[Callable], *_) -> None:
```

Comment:
Is it annoying then, that if another method is added by say ASM with a different number of args/kwargs, they might have to edit the signature? What if there's an overlap in the key of a kwarg between two methods, but we want the value to be different(seems very niche but just curious).

Reply:
> Is it annoying then, that if another method is added by say ASM with a different number of args/kwargs, they might have to edit the signature?

Yeah it does have the potential to cause confusion and make people stumble. To be honest this has been my biggest concern as I move more integrations to use this API and I don't have a great solution for it at the moment.

The fundamental issue is the case where code from two different products responds to the same event and relies on entirely disjoint argument lists to do so. I like the explicitness of having all of the handlers for an event share the same signature, relying on Python's variable parameter lists to denote unused parameters. Maybe the solution is to discourage overloading an event with handlers that don't share any data, instead encouraging the use of multiple events (this also solves ordering dependencies between handlers).

> What if there's an overlap in the key of a kwarg between two methods, but we want the value to be different

It seems ok to me to ask users of this API not to overlap kwarg names, because of how infrequently doing so will create extra work.